docs/dev/agent/building.md

```diff
@@ -0,0 +1,34 @@
+# Building the binary for development
+
+Just like in any other development build of the openshift-install, the process starts with running:
+
+    $ ./hack/build.sh
+
+This will result in the binary being written to *bin/openshift-install*
+
+Since the agent based installer has version / release image checks, though, it is important to patch the binary with the release that you intend to use for your development deployments. If you have not built your own release payload, it is usually a good idea to pick the latest nightly. You can check what the latest nightly is by doing:
+
+    $ RELEASE_VERSION=$(curl -s https://amd64.ocp.releases.ci.openshift.org/api/v1/releasestream/4.12.0-0.nightly/latest | jq -r '.name')
+    $ RELEASE_IMAGE=$(curl -s https://amd64.ocp.releases.ci.openshift.org/api/v1/releasestream/4.12.0-0.nightly/latest | jq -r '.pullSpec')
+
+Once we have these values, we can write them into the binary:
+
+    $ RELEASE_VERSION_LOCATION=$(grep -oba ._RELEASE_VERSION_LOCATION_.XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX bin/openshift-install | cut -d':' -f1)
+    $ RELEASE_IMAGE_LOCATION=$(grep -oba ._RELEASE_IMAGE_LOCATION_.XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX bin/openshift-install | cut -d':' -f1)
+    $ printf "%s\0" "$RELEASE_VERSION" | dd of="bin/openshift-install" bs=1 seek="$RELEASE_VERSION_LOCATION" conv=notrunc
+    $ printf "%s\0" "$RELEASE_IMAGE" | dd of="bin/openshift-install" bs=1 seek="$RELEASE_IMAGE_LOCATION" conv=notrunc
+
+You can verify you got the right patching by running:
+
+    $ ./bin/openshift-install version
```

Comment:
similarly, the intended development workflow AIUI is to set the `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` environment variable rather than editing the binary.